compiler/src/iree/compiler/Dialect/VMVX/vmvx.imports.mlir

```diff
@@ -491,4 +491,59 @@ vm.import @mmt4d.i8i8i32(
   %flags : i32
 )
 
+//==============================================================================
+// pack ops
+//==============================================================================
+
+vm.import @pack.f32f32(
+  %in_buffer : !vm.buffer,
+  %in_offset : i64,
+  %in_stride0 : i64,
+  %out_buffer : !vm.buffer,
+  %out_offset : i64,
+  %out_stride0 : i64,
+  %in_size0 : i64,
+  %in_size1 : i64,
+  %out_size0 : i64,
+  %out_size1 : i64,
+  %out_size2 : i64,
+  %out_size3 : i64,
+  %padding_value : f32,
```

Comment:
We were chatting about it today, but if you made these all the same and took an i64 + the type field (just another i32 with your `f32 | f32 | 0 | 0` type bits) they could all share the same structure (bitcasting in/out of the i64 for things like f32). Not needed now but would allow you to reduce this set.

Reply:
ok, might do that in the future. Note though : as soon as we pun between types of different widths, e.g. encode a `f32` as a `i64`, this would be leaking endianness, right?   (is the f32 occupying the low 4 bytes or the high 4 bytes) ?  Do we care about the MLIR code doing the `vm.call` being endianness-agnostic? If yes, do we want to just specify that we always encode as little-endian regardless of the host endianness so we don't leak it ?

Reply:
We generally don't care about endianness and just assume little-endian - the vm is assumed to follow host-endianness as are any of the codegen binaries we produce. For a time we were trying to be endian agnostic but it's really hard to do with high performance and not really needed - there's no targets we have that are big-endian or mixed-endian (like little-endian CPU but big-endian DSP/accelerator) as thankfully most of those died off :)

Reply:
Agree about not bothering to support big-endian on an equal footing - any surviving big-endian has learned to get the silicon to handle endianness conversions transparently and efficiently, so it's permanently on them.

But there is still a difference between "follow host-endianness and cross fingers that it's little-endian" and "specify explicitly that interfaces are little-endian, so that if someone ports to a big-endian target in the future, it'll be an impl problem on their side to deal with that". I hope we are doing the latter?